### DISCUSSION TITLE
Is there a way to prevent or discourage the use of SuppressWarnings?

### DISCUSSION THREAD

**Discussion by rsoesemann** [pmd/pmd#4223] [Q&A] [answered]
Instead of manually removing all the suppressions in a codebase I would like to run PMD and ignore them.

**Comment by jsotuyod** [comment, marked as answer]
Depends on what usage of `@SuppressWarnings` you are interested in… if it's for using it to suppress PMD warnings, and you are using the CLI, you can use `--show-suppressed` to ignore all suppressions and show all warnings (this will even report those suppressed by comments such as `// NOPMD`).

If it's the overall usage of `@SuppressWarnings`, even for compiler warnings, you can write a custom rule to flag it, such as `//Annotation[pmd-java:typeIs("java.lang.SuppressWarnings")]`.

Alternatively, you specifically target PMD suppressions through `//Annotation[pmd-java:typeIs("java.lang.SuppressWarnings")][descendant::MemberValuePair/StringLiteral[@ConstValue = "PMD"]]`, but beware, PMD honors some standard usages such as `@SuppressWarnings("unused")` that would be missed this way.

**Comment by adangel** [comment]
Note, there are the following two feature requests on this topic:
* #2736
* #2896
